lib/reader/src/parser.rs

```diff
@@ -40,11 +40,32 @@ pub fn parse_test<'a>(text: &'a str) -> Result<TestFile<'a>> {
     let mut parser = Parser::new(text);
     // Gather the preamble comments as 'Function'.
     parser.gather_comments(AnyEntity::Function);
+
+    let commands = parser.parse_test_commands();
+    let mut isa_spec = parser.parse_isa_specs()?;
+
+    if let isaspec::IsaSpec::Some(ref mut isa_vec) = isa_spec {
+        if isa_vec.len() == 1 {
+            // Remove it from the isa_vec just while we parse functions, avoiding a .clone()
+            parser.target_isa = Some(isa_vec.remove(0));
```

Comment:
The parser can just borrow the `TargetIsa` as a reference instead of owning a box. Then you don't need to move it around.